src/Microsoft.Health.Api/Features/HealthChecks/CachedHealthCheck.cs

```diff
@@ -35,52 +35,64 @@ internal sealed class CachedHealthCheck : IHealthCheck, IDisposable
             _healthCheckFactory = EnsureArg.IsNotNull(healthCheckFactory, nameof(healthCheckFactory));
             _options = EnsureArg.IsNotNull(options, nameof(options));
             _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+            _semaphore = new SemaphoreSlim(_options.MaxRefreshThreads, _options.MaxRefreshThreads);
         }
 
         public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
-            DateTimeOffset currentTime = Clock.UtcNow;
-            if (IsCacheFresh(currentTime))
+            CacheSnapshot current = GetCacheSnapshot();
```

Comment:
Sorry if I am missing some pre-context here. But I was expecting something like

return GetHealthCheckResult();

Basically the understanding of caching and refreshing is done in GetHealthCheckResult()
{
   var result = _cachedResult;

   If (result == null || result.IsExpired)
   {
       RefreshCache()
       result = _cachedResult[]
   }
  return result.Result;
}

Reply:
I attempted to simply it a bit more